src/main/java/appeng/core/AELog.java

```diff
@@ -19,93 +19,208 @@
 	{
 		if( AEConfig.instance == null || AEConfig.instance.isFeatureEnabled( AEFeature.Logging ) )
 		{
-			FMLRelaunchLog.log( "AE2:" + ( Platform.isServer() ? "S" : "C" ), level, format, data );
+			final String formattedMessage = String.format( message, params );
+			final Logger logger = getLogger();
+
+			logger.log( level, formattedMessage );
 		}
 	}
 
-	public static void grinder( final String o )
+	/**
+	 * Log an exception with a custom message formated via {@link String#format(String, Object...)}
+	 *
+	 * Similar to {@link AELog#log(Level, String, Object...)}.
+	 *
+	 * @see AELog#log(Level, String, Object...)
+	 *
+	 * @param exception
+	 * @param message the message to be formatted.
+	 * @param params the parameters used for {@link String#format(String, Object...)}.
+	 */
+	public static void log( @Nonnull final Throwable exception, @Nonnull String message, final Object... params )
 	{
-		if( AEConfig.instance.isFeatureEnabled( AEFeature.GrinderLogging ) )
+		if( AEConfig.instance.isFeatureEnabled( AEFeature.Logging ) )
 		{
-			log( Level.DEBUG, "grinder: " + o );
+			final String formattedMessage = String.format( message, params );
+			final Logger logger = getLogger();
+
+			logger.log( Level.WARN, formattedMessage, exception );
 		}
 	}
 
-	public static void integration( final Throwable exception )
+	/**
+	 * @see AELog#log(Level, String, Object...)
+	 * @param format
+	 * @param params
+	 */
+	public static void info( @Nonnull final String format, final Object... params )
 	{
-		if( AEConfig.instance.isFeatureEnabled( AEFeature.IntegrationLogging ) )
-		{
-			error( exception );
-		}
+		log( Level.INFO, format, params );
 	}
 
-	public static void error( final Throwable e )
+	/**
+	 * @see AELog#log(Level, String, Object...)
+	 * @param format
+	 * @param params
+	 */
+	public static void warning( @Nonnull final String format, final Object... params )
 	{
-		if( AEConfig.instance.isFeatureEnabled( AEFeature.Logging ) )
+		log( Level.WARN, format, params );
+	}
+
+	/**
+	 * @see AELog#log(Level, String, Object...)
+	 * @param format
+	 * @param params
+	 */
+	public static void error( @Nonnull final String format, final Object... params )
+	{
+		log( Level.ERROR, format, params );
+	}
+
+	/**
+	 * @see AELog#log(Level, String, Object...)
+	 * @param format
+	 * @param data
+	 */
+	public static void debug( @Nonnull final String format, final Object... data )
+	{
+		if( AEConfig.instance.isFeatureEnabled( AEFeature.DebugLogging ) )
 		{
-			severe( "Error: " + e.getClass().getName() + " : " + e.getMessage() );
-			e.printStackTrace();
+			log( Level.DEBUG, format, data );
 		}
 	}
 
-	public static void severe( final String format, final Object... data )
+	/**
+	 * Use this to log caught {@link Exception} inside a try/catch block when necessary.
+	 *
+	 * Mostly intended for ignored exceptions.
+	 *
+	 * @param exception
+	 */
+	public static void catching( @Nonnull final Throwable exception )
```

Comment:
When actually looking at the majority of the current `AELog.error()` uses, it is really just about log that exception for debug purposes. Rarely the catch block even tries to recover from it.
Thus debug is in these cases probably sufficient and `error()` should be called `exception()` since the beginning.

If we log any exception without any futher information for a reader outside a debug environment, I would say it is already a bad habit and should be reworked. There should be at least one other useful message similar to the change for #2009.
Best case would be in my opinion, log a meanigful message about what went wrong (should the user actually be able to read it) and have the stacktrace only as additional output in debug mode.
If it is a one time error and not reproduceable, we probably cannot even do anything with a stacktrace, otherwise we can simply verify it locally or ask the user to activate debug logging and post the verbose log.